File: Proyecto/assets/js/purchase.js
$(document).ready(() => {

    $("#createPurchase, #createPurchaseProductPaypal, #createPurchaseProductBank").click((event) => {
        // createPurchase();
        completePurchase();
    });
});

const completePurchase = () => {
    const queryString = window.location.search;
const urlParams = new URLSearchParams(queryString);
const productID = urlParams.get('id')
sessionActive = JSON.parse(localStorage.getItem("user"));
const amount = 600
var productPriceText = document.getElementById('product-price').innerText;
var match = productPriceText.match(/\$([0-9.]+)/);
var productPrice = match ? parseFloat(match[1]) : null;

    const body = {
        CustomerName: sessionActive.FirstName,
        Email: sessionActive.Email,
        CustomerState: sessionActive.State,
        CustomerCanton: sessionActive.Canton,
        IdItem: productID,
        creditCard: $('#creditCardNumber').val(),
        monthExpiration: $('#monthExpiration').val(),
        yearExpiration: $('#yearExpiration').val(),
        cvv: $('#cvv').val(),
        amount: amount,
        productCharge: productPrice
    }
    console.log('Datos a enviar:', body);
    axios.post('http://localhost:4001/complete-purchase',body)
        .then(response => {
            if (response.data === true) {
                createPurchase();
                Swal.fire({
                    icon: 'success',
                    iconColor: '#228B22',
                    title: `Your purchase has been created.`,
                    confirmButtonText: 'Continue',
                    text: `Check your email to see our agents contacting you`,
                    confirmButtonColor: '#228B22',
                });
            } else {
                Swal.fire({
                    icon: 'error',
                    iconColor: '#FF0000',
                    title: `Unable to complete the purchase.`,
                    confirmButtonText: 'Continue',
                    text: `Please fill all the fields, verify your credit card and its amount able.`,
                    confirmButtonColor: '#FF0000',
                })
            }
        })
        .catch(error => {
        console.error(error);
        });
}

const queryString = window.location.search;
const urlParams = new URLSearchParams(queryString);
const productID = urlParams.get('id')
//obtener la id
const createPurchase = () => {
    sessionActive = JSON.parse(localStorage.getItem("user"));
    const body = {
        CustomerName: sessionActive.FirstName,
        Email: sessionActive.Email,
        CustomerState: sessionActive.State,
        CustomerCanton: sessionActive.Canton,
        IdItem: productID 
    };

    axios.post('http://localhost:4001/create-order', body)
        .catch((error) => {
            Swal.fire({
                icon: 'error',
                iconColor: '#FF0000',
                title: `Unable to create a purchase.`,
                confirmButtonText: 'Continue',
                text: `Please fill all the fields and try again.`,
                confirmButtonColor: '#FF0000',
            });
        });
};

File: Proyecto/assets/python/purchase.py
'''

from flask import Flask, request, jsonify
from flask_cors import CORS
from datetime import datetime

app = Flask(__name__)
CORS(app)

def validateCreditCard(cardNumber, expirationMonth, expirationYear, cvv):
    if len(cardNumber) != 16 or not cardNumber.isdigit():#16 digits
        return False
    currentYear = datetime.now().year
    currentMonth = datetime.now().month 
    if not (1 <= int(expirationMonth) <= 12) or int(expirationYear) < currentYear or (int(expirationYear) == currentYear and int(expirationMonth) < currentMonth):
        return False
    if len(cvv) != 3 or not cvv.isdigit():
        return False

    return True

@app.route('/api/complete-purchase', methods=['POST'])
def complete_purchase():
    data = request.json
    credit_card = data.get('creditCard')
    expiration_month = data.get('monthExpiration')
    expiration_year = data.get('yearExpiration')
    cvv = data.get('cvv')
    amount = int(data.get('amount'))
    productCharge = int(data.get('productCharge'))  
    remainingAmount = amount - productCharge
    if remainingAmount < 0:
        return jsonify(False)
    if validateCreditCard(credit_card, expiration_month, expiration_year, cvv):
        return jsonify(True)
    else:
        return jsonify(False)

if __name__ == '__main__':
    app.run(debug=True, host='localhost', port=4001)
    
 '''
